### PR TITLE
Fix bug in batch mapping and get right order for outputs.

### PR DESCRIPTION
**Pull request by madiator**
1. Bug in batch mapping: https://github.com/bespokelabsai/curator/issues/276
We just need to map the response to the right generic request. This is a simple fix (but the bug costed a lot of dollars).
   
2. Ordering wasn't being respected earlier: https://github.com/bespokelabsai/curator/issues/288
We need to keep track of the original row idx and sort by it (and then drop that column).
Added a test that tests both the above.

3. Also minor change: replace the deprecated schema_json() with model_json_schema().

### CODE REVIEW

File: src/bespokelabs/curator/request_processor/base_request_processor.py [hunk 1]
```diff
@@ -398,7 +399,10 @@ def create_dataset_files(
                         f"Some requests do not have responses and require_all_responses is True."
                     )
 
-        return Dataset.from_file(dataset_file)
+        d = Dataset.from_file(dataset_file)
```
Comment (RyanMarten) [inline src/bespokelabs/curator/request_processor/base_request_processor.py:402]: Why do we need to sort?

**Review by RyanMarten — APPROVED** [review]
For the fix, I don't believe we need to sort. 
This may add extra overhead, but is better for the end-user.

File: src/bespokelabs/curator/request_processor/base_request_processor.py (hunk 1 above)
Comment (madiator) [reply src/bespokelabs/curator/request_processor/base_request_processor.py:402]: If I sent requests [req1, req2, req3] (say through a dataset), the ordering of the responses can be anything right now. And varies with each run.
It's a bit confusing, and most people expect the responses to be in the order they sent the queries.

**Comment by madiator** [comment]
The overhead is very little, compared to the network costs. Sorting 10k toy examples takes only 25ms for me.

**Pull request merged**